sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/implementation/http/serializer/HttpResponseDecoder.java

```diff
@@ -32,22 +33,18 @@ public HttpResponseDecoder(ObjectSerializer serializer) {
      *
      * @return The decoded HttpResponse.
      */
-    public HttpDecodedResponse decode(HttpResponse response, HttpResponseDecodeData decodeData) {
+    public HttpDecodedResponse decode(HttpResponse<?> response, HttpResponseDecodeData decodeData) {
         return new HttpDecodedResponse(response, this.serializer, decodeData);
     }
 
     /**
-     * A decorated HTTP response which has subscribable body and headers that supports lazy decoding.
-     *
-     * <p>Subscribing to body kickoff http content reading, it's decoding then emission of decoded object. Subscribing
-     * to header kickoff header decoding and emission of decoded object.
+     * A decorated HTTP response that supports lazy decoding.
      */
-    public static class HttpDecodedResponse implements Closeable {
-        private final HttpResponse response;
+    public static class HttpDecodedResponse extends HttpResponse<Object> {
```

Comment:
Is there a reason that this implements `HttpResponse` now? I'm not against it, just don't fully understand the change.

Reply:
It felt better to treat `HttpDecodedResponse` as a type of `HttpResponse` than to have the former store the latter as a property and use `getSourceResponse()` to access it every time we needed to check a request, status code, or any of its other properties.

Conceptually I also like the idea of having it be a type of `HttpResponse`.